com.unity.render-pipelines.core/Runtime/XR/XRSystem.cs

```diff
@@ -0,0 +1,372 @@
+        public static MSAASamples GetDisplayMSAASamples()
+        {
+            return s_MSAASamples;
+        }
+
+        /// <summary>
+        /// Used by the render pipeline to scale the render target on the XR device.
+        /// </summary>
+        /// <param name="renderScale"></param>
+        public static void SetRenderScale(float renderScale)
+        {
+        #if ENABLE_VR && ENABLE_XR_MODULE
+            SubsystemManager.GetInstances(s_DisplayList);
+
+            foreach (var display in s_DisplayList)
+                display.scaleOfAllRenderTargets = renderScale;
+        #endif
+        }
+
+        /// <summary>
+        /// Used by the render pipeline to initiate a new rendering frame through a XR layout.
+        /// </summary>
+        public static XRLayout NewLayout()
+        {
+            RefreshDeviceInfo();
+
+            if (s_Layout.GetActivePasses().Count > 0)
+            {
+                Debug.LogWarning("Render Pipeline error : XRSystem.EndLayout() was not called!");
+                EndLayout();
+            }
+
+            return s_Layout;
+        }
+
+        /// <summary>
+        /// Used by the render pipeline to complete the XR layout at the end of the frame.
+        /// </summary>
+        public static void EndLayout()
+        {
+            if (dumpDebugInfo)
+                s_Layout.LogDebugInfo();
+
+            s_Layout.Clear();
+        }
+
+        /// <summary>
+        /// Used by the render pipeline to render the mirror view to the gameview, as configured by the XR device.
+        /// </summary>
+        /// <param name="cmd"></param>
+        /// <param name="camera"></param>
+        public static void RenderMirrorView(CommandBuffer cmd, Camera camera)
+        {
+        #if ENABLE_VR && ENABLE_XR_MODULE
+            XRMirrorView.RenderMirrorView(cmd, camera, s_MirrorViewMaterial, s_Display);
+        #endif
+        }
+
+        /// <summary>
+        /// Free up resources used by the system.
+        /// </summary>
+        public static void Dispose()
+        {
+            CoreUtils.Destroy(s_OcclusionMeshMaterial);
+            CoreUtils.Destroy(s_MirrorViewMaterial);
+        }
+
+        // Used by the render pipeline to communicate to the XR device the range of the depth buffer.
+        internal static void SetDisplayZRange(float zNear, float zFar)
+        {
+        #if ENABLE_VR && ENABLE_XR_MODULE
+            if (s_Display != null)
+            {
+                s_Display.zNear = zNear;
+                s_Display.zFar = zFar;
+            }
+        #endif
+        }
+
+        internal static void SetDisplaySync()
+        {
+            // Disable vsync on the main display when rendering to a XR device.
+            // XRTODO : expose public property to control this behavior
+            QualitySettings.vSyncCount = 0;
+
+            // On Android and iOS, vSyncCount is ignored and all frame rate control is done using Application.targetFrameRate.
```

Comment:
the comment doesnt match the code below. Should you also test against iOS platform?

Reply:
will fix